database/sql/headers.go

```diff
@@ -396,6 +423,42 @@ func (h *HeadersDb) GetMerkleRootsConfirmations(
 	return confirmations, nil
 }
 
+// GetHeadersHeightOfLocators returns hash and height from db with given locators.
+func (h *HeadersDb) GetHeadersHeightOfLocators(hashtable []interface{}, hashStop *chainhash.Hash) (bh []*dto.DbBlockHeader, err error) {
```

Comment:
What do you think about splitting this func to two parts?

It may make the GetHeadersHeightOfLocators method cleaner and more focused on executing the query and handling the result.

```go
func BuildHashesQuery(hashtable []interface{}) string {
	query := sqlGetHeadersBeginning
	for i := 1; i < len(hashtable); i++ {
		query += sqlGetHeadersHash
	}
	return query + sqlGetHeadersEnd
}

func (h *HeadersDb) GetHeadersHeightOfLocators(hashtable []interface{}, hashStop *chainhash.Hash) ([]*dto.DbBlockHeader, error) {
	query := BuildHashesQuery(hashtable)
	var headers []*dto.DbBlockHeader
	if err := h.db.Select(&headers, h.db.Rebind(query), hashtable...); err != nil {
		h.log.Error().Err(err).Msg("Failed to get headers by locators")
		return nil, err
	}
	return headers, nil
}
```

Reply:
query construction moved to new function